### PR TITLE
Fix unresponsive UI

### PR DESCRIPTION
**Pull request by ke4**
Changes:
- Revert URL check to the previous code as using a regular expression was too expensive
- Use `ng-template` for else condition

### CODE REVIEW

File: client/src/app/shared/components/data-table/data-table.component.html [hunk 1]
```diff
@@ -21,12 +21,12 @@
...
         <ng-template ngx-datatable-cell-template let-rowIndex="rowIndex" let-value="value" let-row="row">
-          <span *ngIf="!isUrl(value)" title="{{value}}">{{value}}</span>
-          <span *ngIf="isUrl(value)">
-          <a target="_blank" href="{{value}}">
-            <mat-icon matTooltip="See Ingest API JSON" [ngStyle]="{'color': 'lightgrey'}">link</mat-icon>
-          </a>
-        </span>
+          <span *ngIf="isUrl(value); else showAsText">
+            <a target="_blank" href="{{value}}">
+              <mat-icon matTooltip="See Ingest API JSON" [ngStyle]="{'color': 'lightgrey'}">link</mat-icon>
```
Comment (jacobwindsor) [inline client/src/app/shared/components/data-table/data-table.component.html:26]: You don't need to use ngStyle here. Just use normal inline `style="color: lightgrey"`. `ngStyle` is for conditional styling

Comment (aaclan-ebi) [inline client/src/app/shared/components/data-table/data-table.component.html:26]: could we remove the tooltip? i think it's irrelevant with the link 😅

**Review by aaclan-ebi — APPROVED** [review]
Thanks for fixing!

File: client/src/app/shared/components/data-table/data-table.component.ts [hunk 2]
```diff
@@ -66,12 +66,11 @@ export class DataTableComponent implements OnInit, OnDestroy {
...
-    const pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
-      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
-      '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
-      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
-      '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
-      '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
-    return !!pattern.test(value);
+    try {
+      const url = new URL(value);
+    } catch (_) {
```
Comment (jacobwindsor) [inline client/src/app/shared/components/data-table/data-table.component.ts:71]: this is a super neat way of checking for URLs! Tiny (personal choice) thing, I don't think you need the `(_)` at all, `catch { }` works fine and reads a bit better

Comment (jacobwindsor) [reply client/src/app/shared/components/data-table/data-table.component.ts:71]: Also this function is useful in other places, maybe consider extracting it to a `shared.utils` module

**Review by jacobwindsor — CHANGES_REQUESTED** [review]

File: client/src/app/shared/components/data-table/data-table.component.html (hunk 1 above)
Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.html:26]: This was in the original code. I just copied it over. I have not touched it, but I can do that change.

File: client/src/app/shared/components/data-table/data-table.component.ts (hunk 2 above)
Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.ts:71]: The problem of this code is that giving us lots of warnings like this: `sanitizing unsafe URL value OBI:0000415`

Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.ts:71]: But I could not find a working solution with sanitiser :-(

Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.ts]: Extracted

File: client/src/app/shared/components/data-table/data-table.component.html (hunk 1 above)
Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.html]: Fixed as requested ;-)

Comment (ke4) [reply client/src/app/shared/components/data-table/data-table.component.html]: Removed

**Review by jacobwindsor — APPROVED** [review]

**Review by aaclan-ebi — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**